src/main.cpp

```diff
@@ -4982,6 +5065,39 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
     }
 
 
+    else if (strCommand == NetMsgType::GETBLOCKTXN)
+    {
+        BlockTransactionsRequest req;
+        vRecv >> req;
+
+        BlockMap::iterator it = mapBlockIndex.find(req.blockhash);
+        if (it == mapBlockIndex.end() || !(it->second->nStatus & BLOCK_HAVE_DATA)) {
+            Misbehaving(pfrom->GetId(), 100);
+            LogPrintf("Peer %d sent us a getblocktxn for a block we don't have", pfrom->id);
+            return true;
+        }
+
+        if (it->second->nHeight < chainActive.Height() - 10) {
```

Comment:
I don't think this is a good idea.
But if you insist doing that, I think you should not use ' - 10'.

Imagine the following event:

* This peer send a cmpct_blocks at block tip - 10
* New block discovered
* Remote peer send GETBLOCKTXN
* This peer will not respond to it causing a timeout for the remote peer

I think you should ideally remove this condition, which is not a DDOS vector, but if you insist, you should put it at a value like - 15 just to be sure we won't timeout the remote peer.